src/ToolBox/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -2335,50 +2336,86 @@ void MethodContext::recGetReadyToRunHelper(CORINFO_RESOLVED_TOKEN* pResolvedToke
                                            CORINFO_CONST_LOOKUP*   pLookup,
                                            bool                    result)
 {
-    LogError("getReadyToRunHelper NYI");
-
-    // TODO: we need a more sophisticated mapping
-
     if (GetReadyToRunHelper == nullptr)
-        GetReadyToRunHelper = new LightWeightMap<DWORDLONG, DWORD>();
+        GetReadyToRunHelper =
+            new LightWeightMap<Agnostic_GetReadyToRunHelper_TOKENin, Agnostic_GetReadyToRunHelper_TOKENout>();
+
+    CORINFO_LOOKUP_KIND genericLookup;
+    ZeroMemory(&genericLookup, sizeof(genericLookup));
+    if (pGenericLookupKind != 0)
+    {
+        genericLookup = *pGenericLookupKind;
+    }
+    Agnostic_GetReadyToRunHelper_TOKENin  key{*pResolvedToken, genericLookup, id};
+    Agnostic_GetReadyToRunHelper_TOKENout value{*pLookup, result};
 
-    // GetReadyToRunHelper->Add((DWORDLONG)cls, result);
+    GetReadyToRunHelper->Add(key, value);
 }
-void MethodContext::dmpGetReadyToRunHelper(DWORDLONG key, DWORD value)
+
+void MethodContext::dmpGetReadyToRunHelper(Agnostic_GetReadyToRunHelper_TOKENin  key,
+                                           Agnostic_GetReadyToRunHelper_TOKENout value)
 {
-    LogError("getReadyToRunHelper NYI");
+    LogError("dmpGetReadyToRunHelper NYI");
 }
```

Comment:
I do use them sometimes (mcs -dump). The printfs are not that hard/bad to write, normally.